template/registry_date.go

```diff
@@ -0,0 +1,96 @@
+package template
+
+import (
+	"fmt"
+	"time"
+
+	"golang.org/x/text/language"
+)
+
+// dateFunc is the implementation of the date function. Locale-sensitive date formatting.
+var dateRegistryFunc = RegistryFunc{
+	Format: dateFunc,
+}
+
+type dateOptions struct {
+	// (default is system default time zone or UTC)
+	//
+	// NOTE: The option is not part of the default registry.
+	// Implementations SHOULD avoid creating options that conflict with these, but
+	// are encouraged to track development of these options during Tech Preview.
+	TimeZone *time.Location
+	// The predefined date formatting style to use (full, long, medium, short).
+	Style string
+}
+
+// parseDateOptions parses :datetime options.
+func parseDateOptions(options Options) (*dateOptions, error) {
+	errorf := func(format string, args ...any) (*dateOptions, error) {
+		return nil, fmt.Errorf("parse options: "+format, args...)
+	}
+
+	var (
+		opts dateOptions
+		err  error
+	)
+
+	styles := oneOf("full", "long", "medium", "short")
+	if opts.Style, err = options.GetString("style", "short", styles); err != nil {
+		return errorf("%w", err)
+	}
+
+	opts.TimeZone = time.UTC
+
+	if v, ok := options["timeZone"]; ok {
+		switch tz := v.(type) {
+		default:
+			return errorf("want timeZone as string or *time.Location, got %T", v)
+		case *time.Location:
+			opts.TimeZone = tz
+		case string:
+			if opts.TimeZone, err = time.LoadLocation(tz); err != nil {
+				return errorf("load TZ data for %s: %w", tz, err)
+			}
+		}
+	}
+
+	return &opts, nil
+}
+
+func dateFunc(operand any, options Options, locale language.Tag) (any, error) {
+	errorf := func(format string, args ...any) (any, error) {
+		return "", fmt.Errorf("date function: "+format, args...)
```

Comment:
To align text with error messages in other registry functions

```suggestion
		return "", fmt.Errorf("exec date function: "+format, args...)
```